tutorials/gradleTutorial.md

```diff
@@ -0,0 +1,170 @@
+# Gradle Tutorial
+
+Gradle is a _build automation tool_ used to automate build processes. There are many ways of integrating Gradle into a project. In this guide, we will be using the _Gradle wrapper_.
+
+* [Introduction](#introduction)
+* [Adding Gradle Support to Your Project](#adding-gradle-support-to-your-project)
+* [Adding Plugins](#adding-plugins)
+  * [CheckStyle](#checkstyle)
+  * [Shadow](#shadow)
+* [Adding Dependencies](#adding-dependencies)
+  * [JUnit](#junit)
+* [Further Reading](#further-reading)
+
+## Introduction
+
+As a developer, you write a _build file_ that describes the project. A build file mainly consists of _plugins_, _tasks_ and _properties_. 
+
+* **Plugins** extend the functionality of Gradle. For example, the `java` plugin adds support for `Java` projects.
+
+* **Tasks** are reusable blocks of logic.  For example, the task `clean` simply deletes the project build directory. Tasks can be composed of other tasks or be dependent on another task. 
+
+* **Properties** change the behavior of tasks. For instance, `mainClassName` of the `application` plugin is a compulsory property which tells Gradle which class is the entrypoint to your application.
+  As Gradle favors [_convention over configuration_](https://en.wikipedia.org/wiki/Convention_over_configuration), there is not much to you need to configure if you follow the recommended directory structure.
+
+## Adding Gradle Support to Your Project 
+
+1. Pull the branch named `gradle`. Merge it to the `master` branch. This will add the Gradle wrapper to your project.
+   ```
+   git checkout --track origin/gradle
+   git checkout master
+   git merge gradle
+   ```
```

Comment:
Is this needed? I thought this branch has all the gradle stuff already? Applies to a few other places in this file.

Reply:
I'll go prune the tutorial and remove the setting up portions.